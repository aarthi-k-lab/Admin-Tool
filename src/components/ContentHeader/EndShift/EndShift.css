.end-shift.end-shift {
  box-sizing: border-box;
  height: 2.5rem;
  background: #fff;
  border: 2px solid #596feb;
  border-radius: 10px;
  font-style: normal;
  font-weight: 700;
  font-size: 15px;
  line-height: 18px;
  color: #596feb;
  margin-right: 1.5rem;
}

.end-shift-disabled.end-shift-disabled {
  box-sizing: border-box;
  height: 2.5rem;
  background: #fff;
  border: 2px solid #fff;
  border-radius: 10px;
  font-style: normal;
  font-weight: 700;
  font-size: 15px;
  line-height: 18px;
  margin-right: 1.5rem;
}
