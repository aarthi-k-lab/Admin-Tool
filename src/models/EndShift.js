const CLEAR_DASHBOARD_DATA = 'clear-dashboard-data';
const SAVE_DISPOSITION_AND_CLEAR_DASHBOARD_DATA = 'save-disposition-and-clear-dashboard-data';

export default {
  CLEAR_DASHBOARD_DATA,
  SAVE_DISPOSITION_AND_CLEAR_DASHBOARD_DATA,
};
