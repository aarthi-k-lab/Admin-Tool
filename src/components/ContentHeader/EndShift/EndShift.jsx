import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import StopIcon from '@material-ui/icons/Stop';
import './EndShift.css';

const EndShift = ({ disabled, onClick }) => (
  <Button
    className="material-ui-button"
    color="primary"
    disabled={disabled}
    onClick={onClick}
    styleName="end-shift"
    variant="outlined"
  >
    <StopIcon />
    End Shift
  </Button>
);

EndShift.defaultProps = {
  disabled: false,
  onClick: () => {},
};

EndShift.propTypes = {
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
};

export default EndShift;
