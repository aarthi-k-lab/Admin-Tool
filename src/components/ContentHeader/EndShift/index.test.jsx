import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import EndShift from '.';

describe('<EndShift />', () => {
  test('snapshot test', () => {
    const tree = renderer
      .create(<EndShift />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('triggers onClick handler', () => {
    const handleClick = jest.fn();
    const wrapper = shallow(<EndShift onClick={handleClick} />);
    wrapper.simulate('click');
    expect(handleClick.mock.calls.length).toBe(1);
  });
});
