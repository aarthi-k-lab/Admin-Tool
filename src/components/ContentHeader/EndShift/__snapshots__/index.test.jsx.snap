// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EndShift /> snapshot test 1`] = `
<button
  className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-textPrimary-4 MuiButton-outlined-9 MuiButton-outlinedPrimary-10 material-ui-button components-ContentHeader-EndShift-EndShift__end-shift--2UamN"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-2"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-30"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M6 6h12v12H6z"
      />
    </svg>
    End Shift
  </span>
  <span
    className="MuiTouchRipple-root-39"
  />
</button>
`;
